### PR TITLE
apply a default initial `forkCount` of `1C` to new plugins to speedup the build.

### PR DESCRIPTION
**Pull request by jtnord**
apply a default initial `forkCount` of `1C` to new plugins to speedup the build.

The number has been chosen from a commonly used number in Jenkinsfils in use on several plugins in the community.

Some are of the opinion that the archetype offers the best practices (they do not exist) and changes to the pipeline should be reflected in all plugins (or the top 200).  I will not be suggesting these changes to any plugins, as the suggestion may be sub-optimal, rather the plugins maintainers should experiment themselves to find and set an optimal value.   Having said this - I still believe it makes sense to advertise (and set a default) for new plugins.

### Testing done

`forkCount` has been used in several plugins already.  this exact Jenkinsfile has not been tested

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

**Review by MarkEWaite — APPROVED** [review]
I like this idea very much.  I agree with the observation that the value is not optimal for a single plugin but is good to include in the archetype to encourage plugin maintainers to consider the potential for test time improvement with `forkCount`.

I ran some comparisons with various plugin builds on ci.jenkins.io and found

* Git plugin and git client plugin are consistently slower with 0.45C than their configured 1C.
* Credentials binding plugin is roughly the same with 0.45C and its not configured forkCount
* Config file provider is consistently slower with 0.45C than with the configured 1C
* Declarative Pipeline plugin is consistently faster with 0.45C than with its not configured forkCount

Past experiments that I've run have been that plugins with few tests don't get much benefit from `forkCount`.  Hopefully, new plugins have enough tests to benefit from `forkCount`

**Review by jglick — APPROVED** [review]
Sounds fine in principle. No opinion on the exact default value—this would have to be arrived at via experimentation. (My personal use of 0.45C is for _local_ tests, not CI.)

**Review by timja — COMMENTED** [review]
All my plugins have performed better with `1C` on CI.

I would have thought better to push harder as it will be cheaper?
Especially for new plugins that will be built with this by default?

**Comment by MarkEWaite** [comment]
> All my plugins have performed better with `1C` on CI.
> 
> I would have thought better to push harder as it will be cheaper? Especially for new plugins that will be built with this by default?

All the plugins that I maintain are using either no `forkCount` or `forkCount: 1C`.  The difference between no value and `1V` has been if there are enough tests that parallel execution actually helps with more than 1-2 minutes of reduced test time.

**Comment by jtnord** [comment]
as per the original summary
> at a value that is most likely to speed up a build once tests are added with the least risk of overwhelming the agent.

I would hope newer plugins start to exercise code using `RealJenkinsRule` more and more and that is what could overload the system.

If the consensue is to make the default 1C and then let people tune it down because of build failures (recalling that these people may not fully understand the implications and why it is failing - e.g. sigkill) and take the risk of increase in any associated helpdesk tickets then that is fine with me - but I was erring on the side of not increasing any load on the helpdesk.

File: common-files/Jenkinsfile [hunk 1]
```diff
@@ -3,6 +3,7 @@
  https://github.com/jenkins-infra/pipeline-library/
 */
 buildPlugin(
+  forkCount: '0.45C', // run this number of tests in parallel for faster feedback.  If the number terminates with a 'C', the value will be multiplied by the number of available CPU cores
```
[the remark this answers is not in the document]
Comment (timja) [reply common-files/Jenkinsfile:6]: I would be in favour of this

[the remark this answers is not in the document]
Comment (jtnord) [reply common-files/Jenkinsfile:6]: given you are involved in the helpdesk (unlike me) if @MarkEWaite is happy to then lets merge this addition and ship

[the remark this answers is not in the document]
Comment (MarkEWaite) [reply common-files/Jenkinsfile:6]: I'm in favor of `1C` as well and am happy to assist people if they encounter a problem with the choice of `1C`.

**Comment by lemeurherve** [comment]
Wouldn't it make more sense to set this value as default in the pipeline library instead of the archetype so it's kept short?
https://github.com/jenkins-infra/pipeline-library/blob/master/vars/buildPlugin.groovy

**Pull request merged**